.github/workflows/build.yml

```diff
@@ -1,7 +1,7 @@
 name: Meson
 
 on:
-  [push, pull_request]
```

Comment:
I don't think this is what you want. Because now pull requests from third-party are not build.

I think what you want is:

```
on:
  pull_request:
  push:
    branches:
      - main
```